packages/core/core-flows/src/order/workflows/claim/confirm-claim-request.ts

```diff
@@ -0,0 +1,361 @@
+import {
+  FulfillmentWorkflow,
+  OrderChangeActionDTO,
+  OrderChangeDTO,
+  OrderClaimDTO,
+  OrderDTO,
+} from "@medusajs/types"
+import { ChangeActionType, Modules, OrderChangeStatus } from "@medusajs/utils"
+import {
+  WorkflowData,
+  createStep,
+  createWorkflow,
+  parallelize,
+  transform,
+  when,
+} from "@medusajs/workflows-sdk"
+import { createRemoteLinkStep, useRemoteQueryStep } from "../../../common"
+import { createFulfillmentWorkflow } from "../../../fulfillment/workflows/create-fulfillment"
+import { createReturnFulfillmentWorkflow } from "../../../fulfillment/workflows/create-return-fulfillment"
+import { previewOrderChangeStep } from "../../steps"
+import { confirmOrderChanges } from "../../steps/confirm-order-changes"
+import { createOrderClaimItemsStep } from "../../steps/create-claim-items"
+import { createReturnItemsStep } from "../../steps/create-return-items"
+import {
+  throwIfIsCancelled,
+  throwIfOrderChangeIsNotActive,
+} from "../../utils/order-validation"
+
+type WorkflowInput = {
+  claim_id: string
+}
+
+const validationStep = createStep(
+  "validate-confirm-claim-request",
+  async function ({
+    order,
+    orderChange,
+    orderClaim,
+  }: {
+    order: OrderDTO
+    orderClaim: OrderClaimDTO
+    orderChange: OrderChangeDTO
+  }) {
+    throwIfIsCancelled(order, "Order")
+    throwIfIsCancelled(orderClaim, "Claim")
+    throwIfOrderChangeIsNotActive({ orderChange })
+  }
+)
+
+function prepareFulfillmentData({
```

Comment:
thought: Think this is 1:1 with the other `prepareFulfillmentData` – if so, we could extract to a utility